src/Microsoft.AspNet.Mvc.Core/Controller.cs

```diff
@@ -1026,6 +1026,61 @@ public virtual Task<bool> TryUpdateModelAsync<TModel>([NotNull] TModel model)
                 predicate);
         }
 
+        /// <summary>
+        /// Validates the specified <paramref name="model"/> instance using values from the controller's current
+        /// <see cref="IValueProvider"/>.
+        /// </summary>
+        /// <param name="model">The model instance to update.</param>
+        /// <returns>A <see cref="bool"/> that on completion returns <c>true</c> if the <see cref="ModelState" />
+        /// is valid; <c>false</c> otherwise. </returns>
+        [NonAction]
+        public virtual bool TryValidateModel([NotNull] object model)
+        {
+            return TryValidateModel(model, prefix: null);
+        }
+
+        /// <summary>
+        /// Validates the specified <paramref name="model"/> instance using values from the controller's current
+        /// <see cref="IValueProvider"/> and a <paramref name="prefix"/>.
+        /// </summary>
+        /// <param name="model">The model instance to update.</param>
+        /// <param name="prefix">The prefix to use when looking up values in the current <see cref="IValueProvider"/>
+        /// </param>
+        /// <returns>A <see cref="bool"/> that on completion returns <c>true</c> if the <see cref="ModelState" />
+        /// is valid; <c>false</c> otherwise. </returns>
+        [NonAction]
+        public virtual bool TryValidateModel([NotNull] object model, string prefix)
+        {
+            if (BindingContext == null)
+            {
+                var message = Resources.FormatPropertyOfTypeCannotBeNull(
+                    nameof(BindingContext),
+                    typeof(Controller).FullName);
+                throw new InvalidOperationException(message);
+            }
+
+            var modelMetadata = MetadataProvider.GetMetadataForType(
+               modelAccessor: null,
+               modelType: model.GetType());
+
+            modelMetadata.Model = model;
+
+            var validationContext = new ModelValidationContext(
+                MetadataProvider,
+                BindingContext.ValidatorProvider,
+                ModelState,
+                modelMetadata,
+                containerMetadata: null);
+
+            var modelName = prefix ?? string.Empty;
+
+            var validationNode = new ModelValidationNode(modelMetadata, modelName);
+            validationNode.ValidateAllProperties = true;
+            validationNode.Validate(validationContext);
+
+            return ModelState.IsValid;
```

Comment:
Was this the behavior in Mvc 5? If the ModelState was already invalid due to another model, this would return false for valid models